src/Inventory/Request.php

```diff
@@ -319,6 +319,18 @@ public function contact($data)
         //For the moment it's the Agent who informs us about the active tasks
         $raw_data = $this->inventory->getRawData();
         if ($raw_data !== null && property_exists($raw_data, 'enabled-tasks')) {
+            $taskneededinv = [
+                'esx',
+                'netdiscovery',
+                'netinventory',
+                'remoteinventory'
+            ];
+            if (
+                !empty(array_intersect($raw_data->{'enabled-tasks'}, $taskneededinv)) &&
+                !in_array('inventory', $raw_data->{'enabled-tasks'})
+            ) {
+                $response['tasks']['inventory'] = $this->handleTask('inventory');
+            }
             foreach ($raw_data->{'enabled-tasks'} as $task) {
```

Comment:
Adding `inventory` to the list of enabled tasks would ensure that it is handled the same way as the others tasks.

```suggestion
            $enabled_tasks = $raw_data->{'enabled-tasks'};

            // The following tasks depends on inventory.
            // When they are enabled, we assume that inventory is enabled.
            $taskneededinv = [
                'esx',
                'netdiscovery',
                'netinventory',
                'remoteinventory',
            ];
            if (
                !empty(array_intersect($enabled_tasks, $taskneededinv)) &&
                !in_array('inventory', $enabled_tasks)
            ) {
                $enabled_tasks[] = 'inventory';
            }

            foreach ($enabled_tasks as $task) {
```